2009-04-16  Michael Tautschnig  <mt@debian.org>

	* fai-savelog: Fixed cd to wrong dir when $LOGREMOTEDIR was set (thanks Robin
		Powell for the patch) (closes: #515704)
Index: trunk/lib/fai-savelog
===================================================================
--- trunk.orig/lib/fai-savelog
+++ trunk/lib/fai-savelog	
@@ -58,18 +58,21 @@
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 save_log_remote_shell() {
 
-    local thislog
+    local thislogdir
     if [ -n "$LOGREMOTEDIR" ];then
-	thislog=$LOGREMOTEDIR/$HOSTNAME/$FAI_ACTION-$fai_rundate
+	thislog=$LOGREMOTEDIR/$HOSTNAME
     else
-	thislog=$HOSTNAME/$FAI_ACTION-$fai_rundate
+	thislog=$HOSTNAME
     fi
 
+    local thislog
+    thislog=$thislogdir/$FAI_ACTION-$fai_rundate
+
     echo "Save log files via $remotesh to $LOGUSER@$LOGSERVER:$thislog"
     find $LOGDIR -size 0 -type f -exec rm {} \;
     $remotesh -l $LOGUSER $LOGSERVER " \
        mkdir -p $thislog ;\
-       cd $HOSTNAME ;\
+       cd $thislogdir ;\
        rm -f last-$FAI_ACTION ;\
        ln -snf $FAI_ACTION-$fai_rundate last-$FAI_ACTION ;\
        ln -snf $FAI_ACTION-$fai_rundate last"
